src/app/deals/[id]/page.tsx

```diff
@@ -10,6 +10,18 @@ type Props = {
   searchParams: { [key: string]: string | string[] | undefined }
 }
 
+export async function getStaticProps() {
```

Comment:
This solution should use the `generateStaticParams` function. You can find the docs here, https://nextjs.org/docs/app/api-reference/functions/generate-static-params.

Reply:
@jamesqquick I was able to use the proper function and got it working properly as well as being statically generated. See photo below
<img width="719" alt="Screenshot 2024-07-13 at 8 32 30 AM" src="https://github.com/user-attachments/assets/e332d995-ca15-4e99-8984-bfc5b84542b3">